roaring/roaring.go

```diff
@@ -243,25 +243,25 @@ func (b *Bitmap) RemoveN(a ...uint64) (changed int, err error) {
 	if len(a) == 0 {
 		return 0, nil
 	}
+
+	for _, v := range a {
+		// Apply to the in-memory bitmap.
+		if b.remove(v) {
+			a[changed] = v
+			changed++
+		}
+	}
+
 	if b.OpWriter != nil {
 		op := &op{
 			typ:    opTypeRemoveBatch,
-			values: a,
+			values: a[:changed],
```

Comment:
hmm. okay, so, this creates a subtle semantic change: previously, if the op log write failed, we stopped *without* changing in-memory data. Now we'll be changing the in-memory data, but returning an error and a count of 0. I don't know whether that's a problem, but it feels not-ideal?

Reply:
great point... I felt kind of bad about it, but couldn't think of a good reason why.

What if we went back and unset/reset the bits after a write fail?

Reply:
Hmm. We do have a record of what we changed, and we know whether they were add/remove operations, so we could in fact do that fairly easily -- just do the other operation to the changed values only. Sure, that seems sane.